### PR TITLE
docs: quickstart - Drupal

### PR DESCRIPTION
**Pull request by sixhobbits**
NOTE: Depends on https://github.com/ritza-co/fusionauth-quickstart-php-drupal-web being moved to the FusionAuth org

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205273462533622
  - https://app.asana.com/0/0/1205877475892192

### CODE REVIEW

File: astro/src/content/quickstarts/quickstart-php-drupal-web.mdx [hunk 1]
```diff
@@ -0,0 +1,752 @@
...
+- `changebank.routing.yml`
+- `UserEmailBlock.php`
+- `UserLoginLogoutBlock.php`
+- `changebank.module`
+
+#### Customize The Changebank Form
+
+In your text editor, select the `ChangebankForm.php` file located at `your-application/web/modules/changebank/src/Form/ChangebankForm.php`. Clear the existing boilerplate contents and add the following code.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/web/modules/custom/changebank/src/Form/ChangebankForm.php"}
```
Comment (vcampitelli) [inline astro/src/content/quickstarts/quickstart-php-drupal-web.mdx:458]: I think we could add 
```php
declare(strict_types = 1);
```
to every PHP file

File: astro/src/content/quickstarts/quickstart-php-drupal-web.mdx [hunk 2]
```diff
@@ -0,0 +1,752 @@
...
+            lang="php" />
+
+#### Customize The Controller
+
+In your text editor, select the `ChangebankController.php` file located at `your-application/web/modules/changebank/src/Controller/ChangebankController.php` and replace the content with the following code.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/web/modules/custom/changebank/src/Controller/ChangebankController.php"}
```
Comment (vcampitelli) [inline astro/src/content/quickstarts/quickstart-php-drupal-web.mdx:465]: The first time we enter this page (just after logging in), `\Drupal::state()->get('changebank_form_value')` returns `NULL`, which raises an error when accessing `$value->getArguments()`.

I opened a PR at https://github.com/ritza-co/fusionauth-quickstart-php-drupal-web/pull/1

**Review by vcampitelli — CHANGES_REQUESTED** [review]
I still need to

File: astro/src/content/quickstarts/quickstart-php-drupal-web.mdx [hunk 3]
```diff
@@ -0,0 +1,752 @@
...
+In the templates directory, create the following files:
+
+- `page--front.html.twig`
+- `block--system-branding-block.html.twig`
+- `region--secondary-menu.html.twig`
+
+The `page--front.html.twig` file uses an image that is currently not in your application. To copy the image to your application, run the following command from `your-application`.
+
+```shell
+cp ../complete-application/web/sites/default/files/money-new.jpg  web/sites/default/files/money-new.jpg
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-php-drupal-web.mdx]: fixed

File: astro/src/content/quickstarts/quickstart-php-drupal-web.mdx (hunk 1 above)
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-php-drupal-web.mdx:473]: added strict type checking

File: astro/src/content/quickstarts/quickstart-php-drupal-web.mdx [hunk 4]
```diff
@@ -0,0 +1,752 @@
...
+
+### Test Authentication
+
+Now you can test the authentication flow.
+
+In the browser, navigate to http://localhost/user/logout to log out of the Drupal application. Navigate to http://localhost/user/login and click on the <Uielement label="Login with generic"/> button.
+
+You should be taken to the FusionAuth login page. Log in with the username `richard@example.com` and password `password`.
+
+You should be redirected back to the Drupal application once logged in.
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-php-drupal-web.mdx:246]: Fixed. Was not setting the username in the kickstart which was causing the plugin to generate a random user name.

File: astro/src/content/quickstarts/quickstart-php-drupal-web.mdx [hunk 5]
```diff
@@ -0,0 +1,752 @@
...
+In this quickstart guide you will build a Drupal web application and integrate it with FusionAuth.
+
+You'll be building it for [ChangeBank](https://www.youtube.com/watch?v=CXDxNCzUspM), a global leader in converting dollars into coins. It'll have areas reserved for users who have logged in as well as public facing sections.
+
+The docker compose file and source code for a complete application are available at
+https://github.com/FusionAuth/fusionauth-quickstart-php-drupal-web
+
+## Prerequisites
+
+- [Composer](https://getcomposer.org/).
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-php-drupal-web.mdx]: updated the Prerequisites

File: astro/src/content/quickstarts/quickstart-php-drupal-web.mdx [hunk 6]
```diff
@@ -0,0 +1,752 @@
...
+<RemoteCode url={frontmatter.coderoot + "/complete-application/web/modules/custom/changebank/changebank.module"}
+            lang="php" />
+
+#### Enable The Changebank Module
+
+To enable the Changebank module, navigate to http://localhost/admin/modules while logged in as admin and select the checkbox next to the Changebank module item. Click the <Uielement label="Install"/> button. You can also enable the module by running the following command in the Drupal Docker container.
+
+```shell
+vendor/bin/drush pm-enable changebank
+```
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-php-drupal-web.mdx]: fixed

File: astro/src/content/quickstarts/quickstart-php-drupal-web.mdx [hunk 7]
```diff
@@ -0,0 +1,752 @@
...
+
+In a terminal, run the following command from the `your-application` directory.
+
+```shell
+composer require 'drupal/openid_connect:3.x-dev@dev' 'drupal/admin_toolbar:^3.4'
+```
+
+Next, enable the modules by visiting the extensions page of your Drupal application at http://localhost/admin/modules. Search for "OpenID Connect" and "Admin Toolbar" using the <Field label="Filter"/> field. Select the module checkboxes and click the <Uielement label="Install"/> button.
+
+Alternatively, you can use Drush to enable the modules. Drush is a command line shell and Unix scripting interface for Drupal, available as a [composer package](https://packagist.org/packages/drush/drush). Install it by running the following command.
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-php-drupal-web.mdx:140]: fixed

File: astro/src/content/quickstarts/quickstart-php-drupal-web.mdx [hunk 8]
```diff
@@ -0,0 +1,752 @@
...
+- <Field label="Site name"/>: `Change Bank`.
+- <Field label="Site email address"/>: `admin@example.com`.
+- <Field label="Username"/>: `admin`.
+- <Field label="Password"/>: `verybadpassword`.
+- <Field label="Email address"/>: `admin@example.com`.
+
+Click <Uielement label="Save and continue"/>.
+
+
+Once you've completed the installation wizard, you should be taken to the welcome page.
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-php-drupal-web.mdx:124]: maybe there is another app listening on `localhost`

File: astro/src/content/quickstarts/quickstart-php-drupal-web.mdx [hunk 9]
```diff
@@ -0,0 +1,752 @@
...
+To set up your Drupal application to use FusionAuth for authentication, head to http://localhost/admin/config/people/openid-connect on your Drupal site.
+
+Add a new client by clicking on <Uielement label="+ Generic OAuth 2.0"/> and fill in the details as follows:
+
+- <Field label="Name"/>: `generic`
+- <Field label="Client ID"/>: `e9fdb985-9173-4e01-9d73-ac2d60d1dc8e`
+- <Field label="Client secret"/>: `super-secret-secret-that-should-be-regenerated-for-production`
+- <Field label="Authorization endpoint"/>: `http://host.docker.internal:9011/oauth2/authorize`
+- <Field label="Token endpoint"/>: `http://host.docker.internal:9011/oauth2/token`
+- <Field label="UserInfo endpoint"/>: `http://host.docker.internal:9011/oauth2/userinfo`
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-php-drupal-web.mdx:215]: Merged,  thanks

**Review by vcampitelli — APPROVED** [review]

**Pull request merged**